internal/search/crawler/github/crawler.go

```diff
@@ -0,0 +1,577 @@
+// Package github implements the crawler.Crawler interface, getting data its
+// from the Github search API.
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"math"
+	"net/http"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/damienr74/kustomize/internal/search/doc"
+)
+
+var logger = log.New(os.Stdout, "Github Crawler: ",
+	log.LstdFlags|log.LUTC|log.Llongfile)
+
+// Implements crawler.Crawler.
+type githubCrawler struct {
```

Comment:
Can you move split this file into another PR? I'll review the query parts in this PR. Thanks.

Reply:
Sure!